### ISSUE TITLE
Add valid labels for search button and input fields (a11y issue)

### ISSUE THREAD

**Issue by batpigandme** [quarto-dev/quarto-cli#678]
Both the collapsed and expanded search features (the button, and then the input field) currently have `aria-labelledby` properties set that reference an element that doesn't exist (and, thus, they have no label)
``` html
<div class="aa-Autocomplete" role="combobox" aria-expanded="false" aria-haspopup="listbox" aria-labelledby="autocomplete-0-label">
<button class="aa-DetachedSearchButton">
...
</button>
</div>
```

There are three associated accessibility issues I get related to the search:

1. ARIA input fields must have an accessible name
    Element: `<div class="aa-Autocomplete" role="combobox" aria-expanded="false" aria-haspopup="listbox" aria-labelledby="autocomplete-0-label">`
2. Buttons must have discernible text
    Element: `<button class="aa-DetachedSearchButton">`
3. ARIA attributes must conform to valid values
    Element: `<div class="aa-Autocomplete" role="combobox" aria-expanded="false" aria-haspopup="listbox" aria-labelledby="autocomplete-0-label">` 
    Error: ARIA attribute element ID does not exist on the page: aria-labelledby="autocomplete-0-label"

The third presumably applies to the search input when opened on the element below:
``` html
<input class="aa-Input" aria-autocomplete="both" aria-labelledby="autocomplete-0-label" id="autocomplete-0-input" autocomplete="off" autocorrect="off" autocapitalize="off" enterkeyhint="search" spellcheck="false" placeholder="" maxlength="512" type="search">
```

As far as internationalization goes, it looks like the buttons have [translation parameters in the Algolia docs](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#parameters).

**Comment by dragonstyle** [comment]
This is injected by our auto-complete library, so the simplest fix would be upstream- related issue:

https://github.com/algolia/autocomplete/issues/963

For now, postponing and I'll work on trying to get fixed upstream.

**Comment by signekb** [comment]
Hi! 
What's the status on this? I'm having the same issue.

**Comment by cderv** [comment]
Upstream issue is still opened, and iIssue here also as we didn't do any specific handling on Quarto side. 

As said, this would need to be dealt upstream in the autocomplete library. 

So work to be done (which is usally the status for issue still opened in this repo)

**Comment by signekb** [comment]
Ah, yeah, makes sense. Thanks for getting back to me so quickly.

**Comment by srvanderplas** [comment]
This is going to be a much bigger issue soon, as WCAG 2.2 guidelines roll out to universities and any page used in any course must be compliant under ADA. I don't have the skills to fix this, but just boosting because I'm auditing my quarto book and finding that the search and sidebar are my biggest issues at the moment.

**Comment by cderv** [comment]
This requires an upstream library fix. It does not seem it gets traction there so we may need to spend some time trying to fix this upstream ourselves... 🤔

**Comment by cscheid** [comment]
We're still dependent on algolia to fix this, but since the library is [open source](), we've decided to attempt a PR there during 1.8.

**Comment by cscheid** [comment]
Closed by https://github.com/quarto-dev/quarto-cli/pull/12730. (We left a PR upstream, but patched our distribution locally.)
